### DISCUSSION TITLE
Problems of running ROSCO in Simulink

### DISCUSSION THREAD

**Discussion by hzzz123** [NatLabRockies/ROSCO#278] [Q&A]
I used' rsoco2.8 \ matlab _ toolbox \ runfast.m' to run the example file' rosco-2.8.0 \ test _ cases \ 5mw _ land _ Simulink \ 5mw _ land _ Simulink.fst' in MATLAB, and the following error occurred:
Misuse of Pre_LoadRotPerf
Error loading file

Error ReadWrite_FAST (line 47)
Cx = Pre_LoadRotPerf(fullfile(fast.FAST_directory,PerfFileName(2:end-1)));

Error runFAST (line 29)
[Param,Cx] = ReadWrite_FAST(fast);

1.What is the reason and how to solve it?
2.Running openfast in matlab/Simulink, what paths need to be added to set the path? Is there an official explanation?
thanks

**Comment by dzalkind** [comment]
The official explanation is [here](https://github.com/NREL/ROSCO/tree/main/Matlab_Toolbox), though these scripts can not be actively supported at this time. If you have updates you can contribute, they would be greatly appreciated by the community.

**Reply by hzzz123** [reply]
Thank you for your reply, which solved some of my problems.

**Reply by Crystal0912** [reply]
Hello,

I encountered the same problem as you. Unfortunately, I have no idea about this after checking other issues and discussions. Could you please tell me how you solved it?

**Comment by yugao0912** [comment]
> Thank you for your reply, which solved some of my problems.



> I used' rsoco2.8 \ matlab _ toolbox \ runfast.m' to run the example file' rosco-2.8.0 \ test _ cases \ 5mw _ land _ Simulink \ 5mw _ land _ Simulink.fst' in MATLAB, and the following error occurred: Misuse of Pre_LoadRotPerf Error loading file
> 
> Error ReadWrite_FAST (line 47) Cx = Pre_LoadRotPerf(fullfile(fast.FAST_directory,PerfFileName(2:end-1)));
> 
> Error runFAST (line 29) [Param,Cx] = ReadWrite_FAST(fast);
> 
> 1.What is the reason and how to solve it? 2.Running openfast in matlab/Simulink, what paths need to be added to set the path? Is there an official explanation? thanks

Hello,

I encountered the same problem as you. Unfortunately, I have no idea about this after checking other issues and discussions. Could you please tell me how you solved it?

Best

**Reply by hzzz123** [reply]
Sorry, I haven't solved it yet. You can replace it with a lower version first.

**Reply by Crystal0912** [reply]
Hello,

I encountered the same problem as you. Unfortunately, I have no idea about this after checking other issues and discussions. Could you please tell me how you solved it?

**Comment by Crystal0912** [comment]
>>ROSCOrun.m
GetFASTPar:: Parameter NTypes not found.
GetFASTPar:: Parameter PerfFileName not found.
错误使用 Pre_LoadRotPerf
Error loading file

出错 ReadWrite_FAST (第 49 行)
Cx   = Pre_LoadRotPerf(fullfile(discon_dir,PerfFileName(2:end-1)));

出错 ROSCOrun_5MW2 (第 29 行)
[Param,Cx] = ReadWrite_FAST(fast);


Hello,
I encountered the same problem as you. Unfortunately, I have no idea about this after checking other issues and discussions. Could you please tell me how you solved it?
